### PR TITLE
[Tests] Fix the template for test generation when there is a CHAR_STRING variable used for a test only cluster

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem
When a `CHAR_STRING` in the `config` section of YAML is used for test only cluster, it does not generates the correct content.

#### Change overview
 * Update template to generate the right content
 
#### Testing
I have locally tested with some local YAML file where I need that.

### CODE REVIEW

File: examples/chip-tool/templates/partials/test_cluster.zapt [hunk 1]
```diff
@@ -245,13 +245,14 @@ class {{filename}}: public TestCommand
...
     {
         SetIdentity(kIdentity{{asUpperCamelCase identity}});
-        return {{command}}({{#chip_tests_item_parameters}}{{#not_first}}, {{/not_first}}{{#if (isString type)}}"{{/if~}}
+        return {{command}}({{#chip_tests_item_parameters}}{{#not_first}},{{/not_first}}
+          {{#*inline "defaultValue"}}{{asTypedLiteral (chip_tests_config_get_default_value definedValue) (chip_tests_config_get_type definedValue)}}{{/inline}}
           {{~#if (chip_tests_config_has definedValue)~}}
-          m{{asUpperCamelCase definedValue}}.HasValue() ? m{{asUpperCamelCase definedValue}}.Value() : {{asTypedLiteral (chip_tests_config_get_default_value definedValue) (chip_tests_config_get_type definedValue)}}
-          {{else}}
-          {{definedValue}}
+          m{{asUpperCamelCase definedValue}}.HasValue() ? m{{asUpperCamelCase definedValue}}.Value() : {{~#if (isString type)}}chip::CharSpan("{{>defaultValue}}", strlen("{{>defaultValue}}")){{else}}{{>defaultValue}}{{/if~}}
```
Comment (bzbarsky-apple) [inline examples/chip-tool/templates/partials/test_cluster.zapt:251]:
```suggestion
          m{{asUpperCamelCase definedValue}}.HasValue() ? m{{asUpperCamelCase definedValue}}.Value() : {{~#if (isString type)}}chip::CharSpan::fromCharString("{{>defaultValue}}"){{else}}{{>defaultValue}}{{/if~}}
```

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/chip-tool/templates/partials/test_cluster.zapt (hunk 1 above)
Comment (vivien-apple) [reply examples/chip-tool/templates/partials/test_cluster.zapt:251]: This is much much prettier. Thanks !

**Comment by vivien-apple** [comment]
It sounds like `restyled` status has not been reported. Let's rebase and see.

**Review by woody-apple — APPROVED** [review]
Fast tracking test tool changes.

**Pull request merged**